core/src/main/java/org/apache/accumulo/core/client/summary/SummarizerConfiguration.java

```diff
@@ -253,7 +253,7 @@ public Builder addOptions(String... keyValuePairs) {
       Preconditions.checkArgument(keyValuePairs.length % 2 == 0 && keyValuePairs.length > 0,
           "Require an even, positive number of arguments, got %s", keyValuePairs.length);
       for (int i = 0; i < keyValuePairs.length; i += 2) {
-        addOption(keyValuePairs[i], keyValuePairs[i + 1]);
+        addOption(keyValuePairs[i], keyValuePairs[i + 1]); // lgtm [java/index-out-of-bounds]
```

Comment:
Could instead add a bounds check here to ensure there's at least 2. An explicit check and throw is better than relying on the incidental error. I think Preconditions might have a convenience method for bounds checking, that throws a reasonable RTE type, with a better message.

Reply:
Any non even number should fail the preconditions check prior to the loop.  I added this comment to ignore the alert as a false positive.  LGTM has an open issue to fix this case: https://discuss.lgtm.com/t/false-positive-java-index-out-of-bounds-with-preconditions-check/1787

Reply:
In that case, you should be able to replace your loop condition with: `i + 1 < keyValuePairs.length`, which should make LGTM happy. Since we've already checked that the amounts are even, the loop condition is equivalent. Alternatively, start the loop at `i=1`, and use `[i]` and `[i-1]` for the indices.

Reply:
FWIW, I think adding the exception comment is fine... but if you can make the code work without needing a comment, I think that's slightly better, even if only slightly.

Reply:
Fixed in 548de10